### DISCUSSION TITLE
trainging, validation and testing datasets

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#2555] [Q&A] [answered]
Hi,
Burn uses huggingface datasets such as mnist, cifar10 and housing price dataset as examples. However, there are only training and testing dataset, do we need validation dataset when we training a model (burn uses testing dataset as validataion dataset, while it's best practice to use separated datasets for testing and validation)?

**Comment by laggui** [comment, marked as answer]
Actually, the housing dataset has a train, validation and test split. But you're correct about the other two.

It's best practice to have a different validation and testing dataset, but in practice not always the case 😅

The testing split could always be split to form a small validation set and new test set.
